src/jrd/pag.cpp

```diff
@@ -266,99 +266,94 @@ namespace
 		CCH_RELEASE(tdbb, &window);
 	}
 
-	ULONG ensureDiskSpace(thread_db* tdbb, WIN* pip_window, const PageNumber pageNum, ULONG pipUsed)
+	ULONG ensureDiskSpace(thread_db* tdbb, WIN* pipWindow, const PageNumber pageNum, const ULONG pipUsed)
 	{
 		const auto* dbb = tdbb->getDatabase();
 		const PageManager& pageMgr = dbb->dbb_page_manager;
 		PageSpace* const pageSpace = pageMgr.findPageSpace(pageNum.getPageSpaceID());
 
-		ULONG newUsed = pipUsed;
 		const ULONG sequence = pageNum.getPageNum() / pageMgr.pagesPerPIP;
-		const ULONG relative_bit = pageNum.getPageNum() - sequence * pageMgr.pagesPerPIP;
+		const ULONG relativeBit = pageNum.getPageNum() - sequence * pageMgr.pagesPerPIP;
+
+		if (relativeBit + 1 <= pipUsed)
+			return pipUsed;
```

Comment:
Condition 
`if (relativeBit + 1 <= pipUsed)`
is equivalent to
`if (relativeBit < pipUsed)`
and after it following assertion seems to make no sense